src/software/sensor_fusion/filter/ball_filter.cpp

```diff
@@ -0,0 +1,363 @@
+#include "software/sensor_fusion/filter/ball_filter.h"
+
+#include <Eigen/Dense>
+#include <algorithm>
+#include <limits>
+
+#include "shared/constants.h"
+#include "software/geom/util.h"
+#include "software/math/math_functions.h"
+#include "software/new_geom/util/closest_point.h"
+
+BallFilter::BallFilter(unsigned int min_buffer_size, unsigned int max_buffer_size)
+    : _min_buffer_size(min_buffer_size),
+      _max_buffer_size(max_buffer_size),
+      ball_detection_buffer(max_buffer_size)
+{
+}
+
+void BallFilter::addNewDetectionsToBuffer(std::vector<BallDetection> new_ball_detections,
+                                          const Field &field)
+{
+    // Sort the detections in increasing order before processing. This places the oldest
+    // detections (with the smallest timestamp) at the front of the buffer, and the most
+    // recent detections (largest timestamp) at the end of the buffer.
+    std::sort(new_ball_detections.begin(), new_ball_detections.end());
+
+    for (const auto &detection : new_ball_detections)
+    {
+        // Ignore any detections that are not anywhere within the field
+        // and ignore any detections that are in the past
```

Comment:
We're not ignoring detections in the past here, but we do below. We should remove that bit of the comment here.

Reply:
removed